src/components/keyboard-aware-container.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import React, { Component } from 'react';
+import { View, Keyboard, Platform } from 'react-native';
+
+type Props = {
+  onKeyboardStateChange: (state: string, event: any) => void,
+  style?: Object | Array<Object>,
+  children: Object,
+};
+
+export class KeyboardAwareContainer extends Component<Props> {
+  constructor(props) {
+    super(props);
+    this.keyboardWillShowSub = null;
+    this.keyboardWillHideSub = null;
+    this.keyboardDidHideSub = null;
+    this.keyboardDidShowSub = null;
+  }
+
+  componentWillMount() {
+    this.keyboardWillShowSub = Keyboard.addListener('keyboardWillShow', e => {
+      if (Platform.OS === 'ios') {
+        this.props.onKeyboardStateChange('show', e);
+      }
+    });
+    this.keyboardWillHideSub = Keyboard.addListener('keyboardWillHide', e => {
+      if (Platform.OS === 'ios') {
+        this.props.onKeyboardStateChange('hide', e);
+      }
+    });
+    this.keyboardDidShowSub = Keyboard.addListener('keyboardDidShow', e => {
+      if (Platform.OS === 'android') {
+        this.props.onKeyboardStateChange('show', e);
+      }
+    });
+    this.keyboardDidHideSub = Keyboard.addListener('keyboardDidHide', e => {
+      if (Platform.OS === 'android') {
+        this.props.onKeyboardStateChange('hide', e);
+      }
+    });
```

Comment:
How about adding listeners by platform? Seems clearer and only 2 listeners at the same time, instead of 4.
```
if (Platform.OS === 'ios') {
    this.keyboardShowListener = Keyboard.addListener(xx, e=> ...);
    this.keyboardHideListener = Keyboard.addListener(xx, e=> ...);
} else if (Platform.OS === 'android') { ... }
```

Reply:
Got it. Will work on that.

Reply:
Curious about different event names for different platforms. Can you explain a bit more?